extensions/s3/src/test/java/io/deephaven/extensions/s3/S3SeekableChannelTestBase.java

```diff
@@ -96,17 +101,31 @@ public int read() {
         }
     }
 
-    private URI uri(String key) {
+    URI uri(String key) {
         return URI.create(String.format("s3://%s/%s", bucket, key));
     }
 
-    private void putObject(String key, RequestBody body) {
+    void putObject(String key, RequestBody body) {
         client.putObject(PutObjectRequest.builder().bucket(bucket).key(key).build(), body);
         keys.add(key);
     }
 
+    void putDirectory(final File directory) {
+        final String directoryName = directory.getName();
+        final Path directoryPath = directory.toPath();
+        try (final Stream<Path> paths = Files.walk(directoryPath)) {
+            paths.filter(Files::isRegularFile)
+                    .forEach(file -> {
+                        final String key = directoryName + "/" + directoryPath.relativize(file);
+                        putObject(key, RequestBody.fromFile(file.toFile()));
+                    });
+        } catch (final IOException e) {
+            throw new UncheckedDeephavenException("Failed to walk directory: " + directory, e);
+        }
+    }
```

Comment:
I've added a more general way to upload directories to S3 using the S3Transfer manager; see https://github.com/deephaven/deephaven-core/pull/5442.